plugins/kafka/kafka.cpp

```diff
@@ -0,0 +1,1118 @@
+
+    //--------------------------------------------------------------------------
+    // Static Methods (internal)
+    //--------------------------------------------------------------------------
+
+    /**
+     * Look for an optional configuration file and apply any found configuration
+     * parameters to a librdkafka configuration object.
+     *
+     * @param   configFilePath      The path to a configuration file; it is not
+     *                              necessary for the file to exist
+     * @param   globalConfigPtr     A pointer to the configuration object that
+     *                              will receive any found parameters
+     * @param   traceLevel          The current log trace level
+     */
+    static void applyConfig(const char* configFilePath, RdKafka::Conf* globalConfigPtr, __int32 traceLevel)
+    {
+        if (configFilePath && *configFilePath && globalConfigPtr)
+        {
+            std::string                 errStr;
+            StringBuffer                fullConfigPath;
+
+            fullConfigPath.append(CONFIG_DIR).append(PATHSEPSTR).append(configFilePath);
+
+            Owned<IProperties>          properties = createProperties(fullConfigPath.str(), true);
+            Owned<IPropertyIterator>    props = properties->getIterator();
+
+            ForEach(*props)
+            {
+                StringBuffer key = props->getPropKey();
+
+                key.trim();
+
+                if (key.length() > 0 && key.charAt(0) != '#')
+                {
+                    if (strcmp(key.str(), "metadata.broker.list") != 0)
+                    {
+                        const char*     value = properties->queryProp(key);
+
+                        if (value && *value)
+                        {
+                            if (globalConfigPtr->set(key.str(), value, errStr) != RdKafka::Conf::CONF_OK)
+                            {
+                                DBGLOG("Kafka: Failed to set config param from file %s: '%s' = '%s'; error: '%s'", configFilePath, key.str(), value, errStr.c_str());
+                            }
+                            else if (traceLevel > 4)
+                            {
+                                DBGLOG("Kafka: Set config param from file %s: '%s' = '%s'", configFilePath, key.str(), value);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        DBGLOG("Kafka: Setting '%s' ignored in config file %s", key.str(), configFilePath);
+                    }
+                }
+            }
+        }
+    }
+
+    //--------------------------------------------------------------------------
+    // Plugin Classes
+    //--------------------------------------------------------------------------
+
+    KafkaStreamedDataset::KafkaStreamedDataset(Consumer* _consumerPtr, IEngineRowAllocator* _resultAllocator, __int32 _traceLevel, __int32 _maxRecords)
```

Comment:
minor: We would generally use int for trace level - where the size of the integer doesn't matter